src/components/swimlanes/swimLanes.tsx

```diff
@@ -0,0 +1,74 @@
+import {
+  Box,
+  Card,
+  CardContent,
+  CardHeading,
+  Flex,
+  Heading,
+  SimpleGrid,
+  Spacer,
+  Text,
+  Link,
+} from "@nypl/design-system-react-components";
+import React from "react";
+
```

Comment:
This where I'd like to use a custom hook, something like this:
`const [lanesWithNumItems, setLanesWithNumItems] = useState([]);

  useEffect(() => {
    async function fetchData() {
      const lanes = data.lanes;
      const updatedLanes = await Promise.all(
        lanes.map(async (lane) => {
          const updatedCollections = await Promise.all(
            lane.collections.map(async (collection) => {
              try {
                const numItems = await getNumItems(collection.uuid);
                return { ...collection, numItems };
              } catch (error) {
                return { ...collection, numItems: 0 };
              }
            })
          );
          return { ...lane, collections: updatedCollections };
        })
      );

      setLanesWithNumItems(updatedLanes);
    }

    fetchData();
  }, []);

  return lanesWithNumItems;
}
`
Maybe bc this is client-side/getting CORS errors, and when the API call is in `getServerSideProps`, it doesn't have that issue?

Reply:
This is a good idea but you're right it's a client-side CORS issue. This makes sense since Repo API should be protected.

Even if Repo API updates and allows this site to make requests, the other issue with this approach is around security. In `getNumItems`, it's getting a value for `process.env.AUTH_TOKEN` and the client side cannot read (intentionally) that server side env variable.

You can create this custom hook if you:

- add back the `/api/getNumItems` endpoint,
- inside that API handler, call `getNumItems(...)`,
- create a new client-side function that makes calls to `/api/getNumItems?uuid=...`. This new function is what you would call here instead of `await getNumItems(collection.uuid)`.

It's not a bad approach but some of my thoughts:
- If you can make the request on the server to get the data and pass it to the client, it'll be faster since the server is faster than the client. When the page loads, you'd be making multiple requests to the internal API endpoint for every item and in turn that will make requests to the Repo API.
- From the previous comment, because there's one parent component between `getServerSideProps` and `SwimLanes`, it seems fine to pass the data down as a prop.
- This approach makes it _slightly_ harder to test. You can test the custom hook in isolation but now the `SwimLanes` component is coupled with the hook. It can be easier to test `SwimLanes` if you can control the test data passed to it.
- This doesn't take into account that you need to render one swim lane, then featured content, then more swim lanes. The custom hook might make it harder to wedge the new component between lanes. This is what I'm thinking but there are definitely more ways to do this:
```
<SwimLanes lanesWithNumItems={lanesWithNumItems[0]} />
<FeaturedContent {...} />
<SwimLanes lanesWithNumItems={lanesWithNumItems[...]} />
```

I definitely think it's worth your time to try the approach and see the pros and cons of it. It'll give you a better idea of API calls and structure for this and future features.